x/slashing/tick.go

```diff
@@ -0,0 +1,54 @@
+package slashing
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/abci/types"
+	crypto "github.com/tendermint/go-crypto"
+)
+
+func NewBeginBlocker(sk Keeper) sdk.BeginBlocker {
+	return func(ctx sdk.Context, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+		// Tag the height
+		heightBytes := make([]byte, 8)
+		binary.LittleEndian.PutUint64(heightBytes, uint64(req.Header.Height))
+		tags := sdk.NewTags("height", heightBytes)
+
+		// Deal with any equivocation evidence
+		for _, evidence := range req.ByzantineValidators {
+			var pk crypto.PubKey
+			sk.cdc.MustUnmarshalBinary(evidence.PubKey, &pk)
+			switch {
+			case bytes.Compare(evidence.Type, []byte("doubleSign")) == 0:
+				sk.handleDoubleSign(ctx, evidence.Height, evidence.Time, pk)
+			default:
+				ctx.Logger().With("module", "x/slashing").Error(fmt.Sprintf("Ignored unknown evidence type: %s", string(evidence.Type)))
+			}
+		}
+
+		// Figure out which validators were absent
+		absent := make(map[string]bool)
```

Comment:
idiomatic Go is to use `map[string]struct{}` here (better to have the empty struct than the bool).

Also, why not use `map[crypto.PubKey]struct{}` ? Would reduce the need to fuss with the string/bytes

Reply:
Switched to `map[crypto.PubKey]struct{}`.